.github/workflows/ci.yml

```diff
@@ -22,3 +24,37 @@ jobs:
         run: |
           git-secrets --register-aws
           git-secrets --scan
+  doxygen:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          path: freertos
+      - name: Install Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7.10'
+      - name: Download tag dependency of coreMQTT-Agent
+        run: |
+          # We don't need to generate the coreMQTT docs, we only need the tag file.
+          # Therefore, we can just download it.
+          mkdir -p freertos/source/dependency/coreMQTT/docs/doxygen/output
+          wget -O freertos/source/dependency/coreMQTT/docs/doxygen/output/mqtt.tag \
+          "https://freertos.org/Documentation/api-ref/coreMQTT/docs/doxygen/output/mqtt.tag"
+      - name: Generate doxygen ZIP
+        uses: FreeRTOS/CI-CD-GitHub-Actions/doxygen@main
+        with:
+          path: ./freertos
+          # List of directories containing libraries whose doxygen output will be generated.
+          libs_parent_dir_path: FreeRTOS-Plus/Source,FreeRTOS-Plus/Source/AWS,FreeRTOS-Plus/Source/Application-Protocols
+          generate_zip: true
+      - name: Upload doxygen artifact if main branch
+        if: success() && github.ref == 'refs/heads/main'
+        uses: actions/upload-artifact@v2
+        with:
+          name: doxygen.zip
+          path: ./doxygen.zip
+          retention-days: 2
```

Comment:
Should we retain the artifacts for a longer time?

Reply:
Good question, I don't have any preference on number of days for retention. We have used 2 days as the retention period in [C-SDK's ci.yml](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/main/.github/workflows/ci.yml#L262), and am keeping the same period.

Reply:
What's the intended use of the artifacts? Do we plan to generate the public documentation from this github action?

Reply:
I guess it's fine, since we could just re-run the job if we need to regenerate the documentation. We need to use the specific revision of each repository to make this build reproducible.

Reply:
>What's the intended use of the artifacts? Do we plan to generate the public documentation from this github action?

Yes